### DISCUSSION TITLE
Declare custom`serverRuntimeConfig` and `publicRuntimeConfig` with getConfig

### DISCUSSION THREAD

**Discussion by GabrielOrtiz-at** [vercel/next.js#40930] [Help]
Hi there,
Is there anyone who might know how I can define custom type declarations for `next/config` when `getConfig()` returns `serverRuntimeConfig` and `publicRuntimeConfig` ?

I'm trying to tell typescript what to expect from calling `getConfig` so that I have strongly typed objects when I use it

```tsx
const { serverRuntimeConfig, publicRuntimeConfig } = getConfig()
```

This is what I have so far but it's not working - typescript is still defaulting to the `next/config` types
i have a `global.d.ts` file where I added
```ts
declare module "next/config" {
  type ConfigTypes = () => {
    serverRuntimeConfig: {
      mySecret: string;
    };
    publicRuntimeConfig: {
      staticFolder: string;
    };
  };

  declare function setConfig(): ConfigTypes;
}
```

i include these types in my tsconfig.json
```js
  "include": [
    "next-env.d.ts",
    "global.d.ts",
    "env.d.ts",
    "src/**/*",
    "src/**/*.json",
    "next-i18next.config.d.ts"
  ],
```

Noob question: do i need to restart typescript somehow when adding types globally? (sorry if it's a stupid question)

This is what I'm getting
![Screen Shot 2022-09-26 at 10 09 00 PM](https://user-images.githubusercontent.com/72769793/192437000-29cf35b6-cc61-4384-a2aa-b85a404f4b38.png)

I would love love love any advice - thanks so much!

**Comment by PumpedSardines** [comment]
Write the declaration file like this instead:
```ts
declare module 'next/config' {
  type ConfigTypes = () => {
    publicRuntimeConfig: {
      staticFolder: string;
    };
    serverRuntimeConfig: {
      mySecret: string;
    };
  };

  declare const getConfig: ConfigTypes;

  export default getConfig;
}
```
That worked for me
